.github/workflows/changelog.yml

```diff
@@ -0,0 +1,33 @@
+# This action requires that any PR targeting the main branch should touch at
+# least one CHANGELOG file. If a CHANGELOG entry is not required, add the "Skip
+# Changelog" label to disable this action.
+
+name: changelog
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, labeled, unlabeled]
+    branches:
+      - main
+jobs:
+  changelog:
+    runs-on: ubuntu-latest
+    if: "!contains(github.event.pull_request.labels.*.name, 'Skip Changelog')"
```

Comment:
Since you can only add labels if you have write permissions, this can be cumbersome for some contributors. I would suggest moving to some field that is modifiable by contributors (so e.g. having a `/skipchangelog` comment, a `[skip changelog]` string on the PR title or some other field that can be edited by any contributor).

Reply:
It hasn't been too much of a problem for us on the Go repo.  Contributors can ask for someone to add the label if they feel it is appropriate and the check is non-blocking in any event.  Allowing anyone to decide that their PR doesn't need a changelog entry seems likely to lead to everyone deciding their PR doesn't need a changelog entry.

Reply:
This was just my feel when contributing to opentelemetry-go; I understand this may not be shared by all (and maybe this reduces the burden on approvers if you think it's likely people will just blindly skip changelog). Let's try this out and see how it works